pkg/edgeview/src/basics.go

```diff
@@ -0,0 +1,665 @@
+		}
+		if !printed {
+			fmt.Printf("  Controller: %s", serverurl)
+		}
+	}
+
+	var release, partition string
+	retbytes, err = ioutil.ReadFile("/run/eve-release")
+	if err == nil {
+		release = string(retbytes)
+	}
+	retbytes, err = ioutil.ReadFile("/run/eve.id")
+	if err == nil {
+		partition = string(retbytes)
+	}
+
+	fmt.Printf("  EVE-OS release %s, %s", release, partition)
+	fmt.Printf("  Edge-View Ver: %s, JWT expires at %v\n",
+		edgeViewVersion, time.Unix(int64(evStatus.ExpireOn), 0).Format(time.RFC3339))
+	hinfo, err := host.Info()
+	if err == nil {
+		loc, _ := time.LoadLocation("UTC")
+		fmt.Printf("  %v(%v), uptime %d (sec) = %d days",
+			time.Now().In(loc).Format(time.RFC3339), loc, hinfo.Uptime, hinfo.Uptime/(3600*24))
+	}
+	fmt.Println()
+	fmt.Println()
+	closePipe(true)
+}
+
+// dynamically install package for uncommon options
+func addPackage(pathName, pkgName string) error {
+	_, err := os.Stat(pathName)
+	if err != nil {
+		if os.IsNotExist(err) {
+			_, err = runCmd("apk add "+pkgName, false)
+			if err != nil {
+				fmt.Printf("%v\n", err)
+				return err
+			}
+		} else {
+			return err
+		}
+	}
+	return nil
+}
+
+func runCmd(cmd string, isPrint bool) (string, error) {
+	var retStr string
+	var retBytes []byte
+	var err error
+
+	retBytes, err = runCmdInFunction(cmd)
+	if err != nil {
+		if !strings.HasSuffix(err.Error(), "status 1") {
+			fmt.Printf("error: %v\n", err)
+		}
+	} else {
+		retStr = string(retBytes)
+		if isPrint {
+			fmt.Println(retStr)
+			closePipe(true)
+		}
+	}
+	return retStr, err
+}
+
+func runCmdInFunction(cmdOpt string) ([]byte, error) {
+	shellcmd := strings.Fields(cmdOpt)
+	if len(shellcmd) < 1 {
+		return nil, fmt.Errorf("shell command invalid\n")
+	}
+	prog := shellcmd[0]
+	args := shellcmd[1:]
+	cmd := exec.Command(prog, args...)
+	stdout, err := cmd.Output()
+	if err != nil {
+		log.Errorf("shell error: %v", err)
+		return nil, err
+	} else {
+		return stdout, nil
+	}
+}
+
+func runPipeCmds(cmd1, cmd2 string) (string, error) {
+	closePipe(false)
+	shellcmd1 := strings.Fields(cmd1)
+	shellcmd2 := strings.Fields(cmd2)
+	if len(shellcmd1) < 1 || len(shellcmd2) < 1 {
+		reOpenPipe(true)
+		return "", fmt.Errorf("shell command invalid\n")
+	}
+
+	c1 := exec.Command(shellcmd1[0], shellcmd1[1:]...)
+	c2 := exec.Command(shellcmd2[0], shellcmd2[1:]...)
+
+	r, w := io.Pipe()
+	c1.Stdout = w
+	c2.Stdin = r
+
+	var b2 bytes.Buffer
+	c2.Stdout = &b2
+
+	c1.Start()
+	c2.Start()
+	go func() {
+		defer w.Close()
+		c1.Wait()
+	}()
+	c2.Wait()
+	out, err := ioutil.ReadAll(&b2)
+	if err != nil {
+		reOpenPipe(true)
+		return "", err
+	}
+	reOpenPipe(true)
+	return string(out), nil
+}
+
+func printColor(msg, color string) {
+	fmt.Printf(color, msg)
+	fmt.Println("")
+}
+
+func printTitle(msg, color string, sendnow bool) {
+	printColor(msg, color)
+	if sendnow {
+		closePipe(true)
+	}
+}
+
+func splitLines(inStr string) []string {
+	outStr := strings.Split(inStr, "\n")
+	n := len(outStr)
+	return outStr[:n-1]
```

Comment:
Sure this n-1 can't reference index -1 which doesn't exist? AFAIK there is a function in the string package to do what you want.